subsys/bluetooth/host/gatt.c

```diff
@@ -267,6 +275,129 @@ static ssize_t cf_write(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 
 	return len;
 }
+
+static u8_t db_hash[16];
+struct k_delayed_work db_hash_work;
+
+struct gen_hash_state {
+	struct tc_cmac_struct state;
+	int err;
+};
+
+static u8_t gen_hash_m(const struct bt_gatt_attr *attr, void *user_data)
+{
+	struct gen_hash_state *state = user_data;
+	struct bt_uuid_16 *u16;
+	u8_t data[16];
+	size_t len;
+
+	if (attr->uuid->type != BT_UUID_TYPE_16)
+		return BT_GATT_ITER_CONTINUE;
+
+	u16 = (struct bt_uuid_16 *)attr->uuid;
+
+	switch (u16->val) {
+	/* Attributes to hash: handle + UUID + value */
+	case 0x2800: /* GATT Primary Service */
+	case 0x2801: /* GATT Secondary Service */
+	case 0x2802: /* GATT Include Service */
+	case 0x2803: /* GATT Characteristic */
+	case 0x2900: /* GATT Characteristic Extended Properties */
+		if (tc_cmac_update(&state->state,
+				   (uint8_t *)&sys_cpu_to_le16(attr->handle),
```

Comment:
Doesn't __bswap_16 return a new value, I would expect the compiler to just store it for me in a unnamed variable, but perhaps using a variable here is clearer. Btw, we should probably have a big endian board to have the CI check these details automatically.

Reply:
As both  sys_cpu_to_le16 and __bswap_16 are macros what will end up happening is that:

`(uint8_t *)&sys_cpu_to_le16(attr->handle)`

becomes:

- LE: `(uint8_t*)&(attr->handle)`
- BE: `(uint8_t*)&((int) ((((attr->handle) >> 8) & 0xff) | (((attr->handle) & 0xff) << 8)))` and gcc gives the error: _lvalue required as unary '&' operand_

I agree regarding having a big-endian board in CI (especially since Oticon is using a big-endian cpu), but this issue could actually just be checked by a big-endian compilation.